Character.cpp

```diff
@@ -86,32 +90,74 @@ int Character::takeDamage(int damage)
     return hitPoints;
 }
 
+int Character::getHP() const { return hitPoints; }
+int Character::getArmorLevel() const { return armor; }
+int Character::getAttackDamage() const { return attackDamage; }
+bool Character::getIsDefending() const { return isDefending; }
+
+const std::vector<std::unique_ptr<Item>>& Character::getHelpfulItems() const { return helpfulItems; }
+const std::vector<std::unique_ptr<Item>>& Character::getDefensiveItems() const { return defensiveItems; }
+
+void Character::boostArmor( int amount )
+{
+    armor += amount;
+    std::cout << getName() << "'s armor level has been boosted to " << armor << std::endl;
+}
+
+void Character::boostHitPoints( int amount )
+{
+    hitPoints += amount;
+    std::cout << getName() << "'s hit point level has been boosted to " << hitPoints << std::endl;
+}
+
+void Character::boostAttackDamage( int amount )
+{
+    attackDamage += amount;
+    std::cout << getName() << "'s attack damage level has been boosted to " << attackDamage << std::endl;
+}
 
 #include <cassert>
+void Character::printStats()
+{
+    std::cout << getName() << "'s stats: " << std::endl;
+    // assert(false);
+    /*
+    make your getStats() use a function from the Utility.h
+    */
+    std::cout << getStats(); 
+    
+    std::cout << std::endl;
+    std::cout << std::endl;
+}
+
 void Character::attackInternal(Character& other)
 {
     if( other.hitPoints <= 0 )
     {
+        
         /*
         When you defeat another Character: 
             a) your stats are restored to their initial value if they are lower than it.
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
+        resetStatLevel(hitPoints, initialHitPoints, 0.1f);
+        resetStatLevel(armor, initialArmorLevel, 0.1f);
+        resetStatLevel(attackDamage, initialAttackDamage, 0.1f);
+        
+        // assert(false);
         std::cout << getName() << " defeated " << other.getName() << " and leveled up!" << std::endl;        
     }
 }
 
-void Character::printStats()
+void Character::resetStatLevel(int& value, std::unique_ptr<int>& initialPtr, float adjustment)
 {
-    std::cout << getName() << "'s stats: " << std::endl;
-    assert(false);
-    /*
-    make your getStats() use a function from the Utility.h
-    */
-    std::cout << getStats(); 
-    
-    std::cout << std::endl;
-    std::cout << std::endl;
-}
\ No newline at end of file
+    if (value < *initialPtr ) { value = *initialPtr; }
+    value *= (1.f + adjustment);
+    initialPtr.reset( new int(value));
+}
```

Comment:
You can make this function simpler if you change the signature to accept two `int&` instances, instead of what you have now. 
That will also move the dereference to the call site. 
This will also eliminate the `reset( new )`
- [ ] revise